web/demos/kachaka_api_web_sample/src/kachakaApi.ts

```diff
@@ -0,0 +1,340 @@
+import { useCallback, useEffect, useState } from 'react'
+
+import { type UnaryCall } from '@protobuf-ts/runtime-rpc'
+
+import { type KachakaApiClient } from './protos/kachaka-api.client'
+// eslint-disable-next-line import/no-duplicates
+import { type GetRequest, type Metadata } from './protos/kachaka-api'
+// eslint-disable-next-line import/no-duplicates
+import type * as pb from './protos/kachaka-api'
+
+function buildGetRequest(cursor: bigint): GetRequest {
+  return {
+    metadata: {
+      cursor,
+    },
+  }
+}
+
+interface GetKachakaStateResponse {
+  metadata?: Metadata
+}
+
+function useKachakaState<
+  ResponseType extends GetKachakaStateResponse,
+  StateType,
+>(
+  client: KachakaApiClient,
+  rpc: (request: GetRequest) => UnaryCall<GetRequest, ResponseType>,
+  filter: (response: ResponseType) => StateType | undefined,
+): StateType | undefined {
+  const [state, setState] = useState<StateType | undefined>(undefined)
+  const [cursor, setCursor] = useState(BigInt(0))
+
+  useEffect(() => {
+    const f = async () => {
+      try {
+        const response = (await rpc(buildGetRequest(cursor)).response) as
+          | ResponseType
+          | undefined
+        if (!response) {
+          return
+        }
+        setState(filter(response))
+        setCursor(response.metadata!.cursor)
+      } catch (e) {
+        console.log(e)
+      }
+    }
+    f()
+  }, [client, cursor, rpc, filter])
+
+  return state
+}
+
+export function useRobotSerialNumber(client: KachakaApiClient) {
+  return useKachakaState<pb.GetRobotSerialNumberResponse, string>(
+    client,
+    (request) => client.getRobotSerialNumber(request),
+    (response) => {
+      return response.serialNumber
+    },
```

Comment:
下の方では↓こう書いているので、ここも揃えたい。
```suggestion
    (response) => response.serialNumber,
```